src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -53,10 +95,129 @@ export class RetireJsEngine implements RuleEngine {
 		return Promise.resolve(retireJsCatalog);
 	}
 
-	// TODO: We need to actually implement this method.
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/require-await, no-unused-vars
 	public async run(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string, string>): Promise<RuleResult[]> {
-		return [];
+		// RetireJS doesn't accept individual files. It only accepts directories. So we need to resolve all of the files
+		// we were given into a directory that we can pass into RetireJS.
+		const tmpDir = await this.createTmpDirWithDuplicatedTargets(target);
+		const invocationArray: RetireJsInvocation[] = this.buildCliInvocations(rules, tmpDir);
+
+		const retireJsPromises: Promise<RuleResult[]>[] = [];
+		for (const invocation of invocationArray) {
+			retireJsPromises.push(this.executeRetireJs(invocation));
+		}
+
+		// We can combine the results into a single array using .reduce() instead of the more verbose for-loop.
+		return (await Promise.all(retireJsPromises)).reduce((acc, r) => [...acc, ...r], []);
```

Comment:
Nitpick - would be helpful to spell out `acc` and `r` to improve code readability.

Reply:
I'll see what I can come up with. Perhaps `all` and `next` instead of `acc` and `r`?